README.md

```diff
@@ -43,59 +58,81 @@ After your Lambda@Edge environment is prepared, install the starter kit:
 
+## Development
+
+### Testing Locally
+
+To avoid the need to upload your Lambda function to AWS every time you make a change, you can test your Lambda function locally using the `AWS SAM` runtime interface client.
+
+1. Install the [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html).
+2. Run `npm run test:viewer_request` to build the lambda and run it locally with the `viewer-request` event.
+
+You can also run lambda handler using `viewer-response`/`origin-request`/`origin-response` events by running respective [commands](#commands).
+
+Additionally you can change the configuration of lambda providing additional parameters to `template.yaml` or by modifying template event files in `lambda` folder.
+
+### Commands
+
+The following commands are available in the starter kit:
+
+- `npm run build` - Builds the TypeScript files in the `src/` directory creating `dist/handler.js` (ignores `src/examples`).
+- `npm run build:lambda` - Builds the Lambda function and creates a `dist/handler.zip` file.
+- `npm run test:viewer_request` - Runs the Lambda function locally with the `lambda/viewer-request.json` event.
+- `npm run test:viewer_response` - Runs the Lambda function locally with the `lambda/viewer-response.json` event.
+- `npm run test:origin_request` - Runs the Lambda function locally with the `lambda/origin-request.json` event.
+- `npm run test:origin_response` - Runs the Lambda function locally with the `lambda/origin-response.json` event.
+- `npm run clean` - Removes the `dist/` directory.
+
+
 ## Additional Resources and Concepts
 
 ### External Data Fetching & Caching
 
-This starter kit uses standard ES7 async and await fetch methods to handle external data fetching. After fetching the Kameleoon client configuration, the SDK caches the client configuration as a JSON object in-memory. However, very large client configurations may cause the method presented in this starter kit to break. If you experience issues with large datafiles breaking in-memory Lambda caching, consider one of the alternative methods of caching with Lambda@Edge outlined in this [external data](https://aws.amazon.com/blogs/networking-and-content-delivery/leveraging-external-data-in-lambdaedge/) article.
+SDK configuration and collected data is stored in-memory in the Lambda function. In some cases, when the data is too large, it may cause the Lambda function to break. We address this issue by giving the developers option to cache the data more efficiently by using AWS services like `S3`, `DynamoDB`, or `CloudFront`.
```

Comment:
The SDK configuration...is to large, the Lambda function may break.  To address this, you can cache the data more efficiently using AWS services like....